v8/credentials/ccache.go

```diff
@@ -37,16 +38,16 @@ type headerField struct {
 	value  []byte
 }
 
-// Credential cache entry principal struct.
-type principal struct {
+// Credential cache entry Principal struct.
+type Principal struct {
```

Comment:
Making the Principal an exported struct is a major change and upstream may not be willing to do that. Not a problem for me, but we may need to change it later.

Reply:
Yes, but on the other hand it is necessary to access the principal from outside the package.